### PR TITLE
PMM-6773 Don't let unregister kubernetes cluster if it contains DB cl…

### PR DESCRIPTION
**Pull request by BupycHuk**
…usters.

[PMM-6773](https://jira.percona.com/browse/PMM-6773)

- [ ] Build: https://github.com/Percona-Lab/pmm-submodules/pull/1211

- [ ] Links to other linked pull requests (optional).
---
- [ ] Tests passed.
- [ ] Feature build pass.
- [ ] (Re)requested review.
- [ ] Fix conflicts with target branch.
- [ ] Update API dependency.
- [ ] Update jira ticket description if needed.
- [ ] Attach screenshots/console output to confirm new behavior to jira ticket, if applicable.

### CODE REVIEW

File: services/management/dbaas/kubernetes_server.go [hunk 1]
```diff
@@ -75,8 +78,36 @@ func (k kubernetesServer) RegisterKubernetesCluster(ctx context.Context, req *db
...
+				},
+			})
+		if err != nil {
+			return err
+		}
+		if len(xtraDBClusters.Clusters) > 0 {
+			return status.Errorf(codes.FailedPrecondition, "Kubernetes cluster %s has XtraDB clusters", req.KubernetesClusterName)
+		}
+
+		psmdbClusters, err := k.dbaasClient.ListPSMDBClusters(ctx, &dbaascontrollerv1beta1.ListPSMDBClustersRequest{
```
Comment (askomorokhov) [inline services/management/dbaas/kubernetes_server.go:99]: This looks like not the best solution to make an external call in a transaction.

**Review by askomorokhov — APPROVED** [review]
Overall looks good to me.

**Comment by askomorokhov** [comment]
@BupycHuk, please check API test - looks like that fails.

File: services/management/dbaas/kubernetes_server.go (hunk 1 above)
Comment (percona-csalguero) [reply services/management/dbaas/kubernetes_server.go:99]: :+1:

**Review by percona-csalguero — COMMENTED** [review]
Agree with Andrii.

**Review by percona-csalguero — APPROVED** [review]

File: services/management/dbaas/kubernetes_server.go [hunk 2]
```diff
@@ -75,12 +82,62 @@ func (k kubernetesServer) RegisterKubernetesCluster(ctx context.Context, req *db
...
 
 	return &dbaasv1beta1.UnregisterKubernetesClusterResponse{}, nil
 }
+
+func accessError(err error) bool {
+	if err == nil {
+		return false
+	}
+	accessErrors := []*regexp.Regexp{
+		regexp.MustCompile(`.*\.percona\.com is forbidden`),
```
Comment (percona-csalguero) [inline services/management/dbaas/kubernetes_server.go:132]: Can we convert them to global vars so it shouldn't be compiled every time?

**Review by percona-csalguero — CHANGES_REQUESTED** [review]

**Review by percona-csalguero — APPROVED** [review]

**Comment by askomorokhov** [comment]
@BupycHuk , could you please resolve conflicts

**Review by percona-csalguero — APPROVED** [review]

**Review by askomorokhov — APPROVED** [review]

**Pull request merged**